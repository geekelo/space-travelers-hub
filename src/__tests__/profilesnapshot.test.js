import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Myprofile from '../components/myprofile/myprofile';

const mockStore = configureMockStore();

describe('Myprofile Component', () => {
  it('renders reserved missions and active rockets', () => {
    const store = mockStore({
      rocket: {
        value: [
          { id: 1, name: 'Falcon 9', active: true },
          // Add more rocket data as needed for testing
        ],
      },
      missions: {
        missions: [
          { key: 1, missionTitle: 'Mission 1', reserved: true },
          // Add more mission data as needed for testing
        ],
      },
    });

    const { asFragment } = render(
      <Provider store={store}>
        <Myprofile />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  // Add more tests as needed for specific behaviors of the Myprofile component
});
