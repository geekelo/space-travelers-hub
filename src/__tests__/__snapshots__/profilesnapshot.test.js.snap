// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Myprofile Component renders reserved missions and active rockets 1`] = `
<DocumentFragment>
  <div
    class="profileContainer"
  >
    <div
      class="listSections"
    >
      <div
        class="titles"
      >
        <h2
          class="titleText"
        >
          My Missions
        </h2>
      </div>
      <ul
        class="itemsSection"
      >
        <li
          class="lists"
        >
          <p />
        </li>
      </ul>
    </div>
    <div
      class="listSections"
    >
      <div
        class="titles"
      >
        <h2
          class="titleText"
        >
          My Rockets
        </h2>
      </div>
      <ul
        class="itemsSection"
      >
        <li
          class="lists"
        >
          <p>
            Falcon 9
          </p>
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
